libminifi/src/core/yaml/YamlConfiguration.cpp

```diff
@@ -762,106 +762,109 @@ void YamlConfiguration::parsePortYaml(YAML::Node *portNode, core::ProcessGroup *
+          if (!processor->setDynamicProperty(propertyName, rawValueString)) {
+            logger_->log_warn("Unable to set the dynamic property %s with value %s", propertyName.c_str(), rawValueString.c_str());
           } else {
-            auto typedValue = propertyValueNode.as<std::string>();
-            coercedValue = typedValue;
-          }
-        } catch (...) {
-          std::string eof;
-          bool exit_on_failure = false;
-          if (configuration_->get(Configure::nifi_flow_configuration_file_exit_failure, eof)) {
-            utils::StringUtils::StringToBool(eof, exit_on_failure);
-          }
-          logger_->log_error("Invalid conversion for field %s. Value %s", myProp.getName(), propertyValueNode.as<std::string>());
-          if (exit_on_failure) {
-            std::cerr << "Invalid conversion for " << myProp.getName() << " to " << defaultType.name() << std::endl;
-          } else {
-            coercedValue = defaultValue;
+            logger_->log_warn("Dynamic property %s with value %s set", propertyName.c_str(), rawValueString.c_str());
           }
         }
-        std::string rawValueString = propertyValueNode.as<std::string>();
-        if (!processor->setProperty(myProp, coercedValue)) {
-          std::shared_ptr<core::Connectable> proc = std::dynamic_pointer_cast<core::Connectable>(processor);
-          if (proc != 0) {
-            logger_->log_warn("Received property %s with value %s but is not one of the properties for %s. "
-                              "Attempting to add as dynamic property.",
-                              propertyName, rawValueString, proc->getName());
-            if (!processor->setDynamicProperty(propertyName, rawValueString)) {
-              logger_->log_warn("Unable to set the dynamic property %s with value %s", propertyName.c_str(), rawValueString.c_str());
-            } else {
-              logger_->log_warn("Dynamic property %s with value %s set", propertyName.c_str(), rawValueString.c_str());
-            }
-          }
-        } else {
-          logger_->log_debug("Property %s with value %s set", propertyName.c_str(), rawValueString.c_str());
-        }
       }
     }
   }
+}
+
+PropertyValue YamlConfiguration::getValidatedProcessorPropertyForDefaultTypeInfo(const core::Property& propertyFromProcessor, const YAML::Node& propertyValueNode) {
+  PropertyValue defaultValue;
+  defaultValue = propertyFromProcessor.getDefaultValue();
+  auto defaultType = defaultValue.getTypeInfo();
+  try {
+    PropertyValue coercedValue = defaultValue;
+    if (defaultType == typeid(int64_t)) {
+      coercedValue = propertyValueNode.as<int64_t>();
+    } else if (defaultType == typeid(int)) {
+      coercedValue = propertyValueNode.as<int>();
+    } else if (defaultType == typeid(bool)) {
+      coercedValue = propertyValueNode.as<bool>();
+    } else {
+      coercedValue = propertyValueNode.as<std::string>();
+    }
+    return coercedValue;
+  } catch (...) {
```

Comment:
I think it would be worthwhile to catch and log `std::exception` compatible exception types.

Reply:
Will add logging.